virtcontainers/qemu.go

```diff
@@ -1567,3 +1571,23 @@ func (q *qemu) cleanup() error {
 
 	return nil
 }
+
+func (q *qemu) pidFile() string {
+	return filepath.Join(store.RunVMStoragePath, q.id, "pid")
+}
+
+func (q *qemu) pid() int {
+	data, err := ioutil.ReadFile(q.pidFile())
```

Comment:
The PID has been stored through its own file? Don't you think we could improve this by having the PID part of the `qemu.state` structure? This would be saved as part of the `state.json` file associated with the VM, and it would avoid yet another file to be created.